src/main/java/team/chisel/ctm/client/util/TextureMetadataHandler.java

```diff
@@ -188,17 +189,19 @@ public void onModelBake(ModelEvent.BakingCompleted event) {
             if (e.getValue() instanceof AbstractCTMBakedModel baked && 
                     baked.getModel() instanceof ModelCTM ctmModel && 
                     !ctmModel.isInitialized()) {
-                Function<Material, TextureAtlasSprite> spriteGetter = (m) -> event.getModelManager().getAtlas(m.atlasLocation()).getSprite(m.texture());
-                var baker = ModelBakerImplAccessor.createImpl(event.getModelBakery(), ($, m) -> spriteGetter.apply(m), e.getKey());
-                ctmModel.bake(baker, spriteGetter, BlockModelRotation.X0_Y0, e.getKey()); 
+                var baker = ModelBakerImplAccessor.createImpl(event.getModelBakery(), ($, m) -> m.sprite(), e.getKey());
+                ctmModel.bake(baker, Material::sprite, BlockModelRotation.X0_Y0, e.getKey());
+                //Note: We have to clear the cache after baking each model to ensure that we can initialize and capture any textures
+                // that might be done by parent models
+                cache.clear();
```

Comment:
This seems like it could potentially add significant load time due to rework, do you know the impact of this?

Reply:
I am not sure of the impact of this though what I will say is that in 1.20 model and texture loading is a lot faster in general, and without this it doesn't actually get all the texture dependencies properly resolved at which point I am not sure what the point in doing the cache clearing stuff we are doing is at all.

My guess is that it is not a major impact just because it only functions on uninitialized ctm models so unless a resource pack or mods are adding tons of them (with complex parents) it won't have much extra overhead. Not to mention this only runs when the models are baking so it won't impact the actual playing at all, just the resource loading.